### ISSUE TITLE
Attendance warning does not process multilang strings

### ISSUE THREAD

**Issue by Kathrin84** [danmarsden/moodle-mod_attendance#302]
Hi Dan,

we've encountered that the new attendance warning message and title do not process multilang strings. 
You can change this by processing the strings passed by the setting through the format_string function.

Please note:
These strings should respect language the user set in his settings and not the system language. So each user can receive the message in his chosen language.

Best, Kathrin

**Comment by danmarsden** [comment]
Thanks @Kathrin84 - can you clarify which message you mean? - the email content is passed through format string although I see the email subject isn't.

**Comment by Kathrin84** [comment]
Hi Dan,

thank you for your response. 
Yes, you're right - I was not precise enough in my issue. Sorry for that. 
The email content is passed through format string, however it does not respect the user defined language, it uses the system language.

I give you an example: 
I used multilang strings for English and German in the subject and in the email content. I set a user's language in his preferences to English.
The result for this user when he received the email:
* The subject contained both, the subject itself in both languages and the multilang tags in plain (= not passed through format_string)
* The content was only displayed in German (= passed through format_string and system language used)

The desired result should be:
* This user should receive the english version of the subject and the content
* Other users that use German as their language should receive the german version

I hope this helps clarifying your question.
Best, Kathrin

**Comment by danmarsden** [comment]
Thanks Kathrin, 

I can't see any existing API's that do this - most of the multi-lang stuff that happens via email comes from lang strings rather than using format_string which doesn't seem to have a way to specify a user or specific language to use.

I'll do some digging to see if there's a way we can support this but let me know if you have any other examples where an email is sent via cron that supports multi-lang teacher defined content.

of course - pull requests and funding are always welcome! - thanks.

**Comment by danmarsden** [comment]
in fact - just spotted how it's done in a few other places and have pushed through a commit that should fix this. Please let me know if it is working for you now.

**Comment by Kathrin84** [comment]
Hi @danmarsden,
thanks for your work on this!

In general your patch is working as intended, however the email subject is always surrounded by a div:
```
<div class="text_to_html">Attendance warning</div>
<div class="text_to_html">Anwesenheitswarnung</div>
```
Best, Kathrin

**Comment by danmarsden** [comment]
Thanks - just pushed through another commit that should fix that too.

**Comment by danmarsden** [comment]
great - thanks for the confirmation!

**Comment by abias** [comment]
Hi @danmarsden ,

I know that the fix for this issue in ccf59f35c6ae219a710c62f5f10272f87dbfb879 was really tiny, but would you mind to release it to the Moodle plugin repo anytime soon so that we can use the version from there again?

Thanks,
Alex
